dashboard/lib/pd/survey_pipeline/survey_pipeline_helper.rb

```diff
@@ -54,4 +207,59 @@ def create_generic_survey_report(context, workers)
       w.process_data context
     end
   end
+
+  # Find all workshops of the same course and facilitated by a facilitator.
+  #
+  # @param fac_ids [Number] valid facilitator id
+  # @param course [String] valid course name
+  #
+  # @return [Array<Number>] list of workshop ids
+  #
+  def find_related_workshops(fac_id, course)
+    return [] unless fac_id && course.present?
+
+    ws_facilitated_query =
+      "SELECT DISTINCT pd_workshop_id FROM pd_workshops_facilitators "\
+      "WHERE user_id = #{fac_id}"
+    ws_facilitated = ActiveRecord::Base.connection.exec_query(ws_facilitated_query).rows.flatten
```

Comment:
I can see the vulnerability here. I have 2 options to fix it

1. Sanitize `fac_id` using [quote](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/Quoting.html#method-i-quote) as suggested [here](https://stackoverflow.com/a/34403240)

2. Doing something similar to `Pd::Workshop.managed_by`
```
  def find_related_workshops(fac_id, course)
    Pd::Workshop.left_outer_joins(:facilitators).
      where(users: {id: fac_id}, course: course).
      distinct.
      pluck(:id)
  end
```
This looks cleaner but one thing bothers me is it actually joins 3 tables instead of 2, `pd_workshops`, `pd_workshop_facilitators` and `users`. The last join with `users` is only to use `users.id` instead of `pd_workshop_facilitators.user_id` in filter.

The behind-the-scene SQL command is
```
SELECT DISTINCT `pd_workshops`.`id` FROM `pd_workshops` LEFT OUTER JOIN `pd_workshops_facilitators` ON `pd_workshops_facilitators`.`pd_workshop_id` = `pd_workshops`.`id` LEFT OUTER JOIN `users` ON `users`.`id` = `pd_workshops_facilitators`.`user_id` AND `users`.`deleted_at` IS NULL WHERE `pd_workshops`.`deleted_at` IS NULL AND `users`.`id` = 124 AND `pd_workshops`.`course` = 'CS Fundamentals'
```

I can go with option 2 to make the code more succinct, but I'm still interested in learning about better options.

Reply:
1 is fine with me - I just want to make sure we're sanitizing our parameters.